compiler/src/arch_full.ml

```diff
@@ -35,6 +36,7 @@ module type Core_arch = sig
   val known_implicits : (Name.t * string) list
 
   val is_ct_asm_op : asm_op -> bool
+  val is_doit_asm_op : asm_op -> bool
   val is_ct_asm_extra : extra_op -> bool
```

Comment:
I think is_doit_asm_extra need to be added too. 
asm_extra are then compiled into sequence of asm instruction, but this is done at during the last pass of the compilation.

Reply:
Hmm good point, I will have to look at that more. Couldn't the solution also be to run the CT checker after that compiler pass, so there are no asm_extra ops? If the transformation of asm_extra -> sequence of asm instructions is non-trivial then this `is_doit_asm_extra` would have to essentially do the job ofthe CT checker for that sequence. I could imagine that some asm_extra turns into a sequence that contains a mix of DOIT and non-DOIT instructions, but secrets only get passed to one of the two classes or both.

Anyway, I will investigate more.

Reply:
Added this now. I manually investigated the cases of extra OPs and they seem OK, but this will need to be done again wen new are introduced.